### ISSUE TITLE
LabeledStatementCodeFixProvider(CS0164) deletes child node aswell as label.

### ISSUE THREAD

**Issue by jamesHargreaves12** [JosefPihrt/Roslynator#1032]
Fix here https://github.com/JosefPihrt/Roslynator/pull/1031

Also contains a test to make it clear what the issue is.

**Comment by JosefPihrt** [comment]
Hi,

can you clarify why do you think that statements inside the labeled statement should not be deleted when the label is removed?

I would say that it make more sense to remove statements as well because statements are executed only if the label is used.

**Comment by jamesHargreaves12** [comment]
Apologies, for not being clearer in the description. 

The CS0164 Removes labels that are not used via a goto instruction rather than implying that the statement that follows the label is unreachable. See https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0164 

Taking the toy example from the test that I provided.
```
int M()
{
  start:
      return 1;
}
```
Prior to this PR Roslynator would replaced this with:
```
int M()
{
}
```

Whereas I believe the correct fix is:
```
int M()
{
    return 1;
}
```

**Comment by jamesHargreaves12** [comment]
@JosefPihrt Sorry for the ping but it has been a month so I just wanted to check that this one hasn't slipped through the net.

**Comment by JosefPihrt** [comment]
I still think that it is reasonable to remove the child statement of the labeled statement (i.e. as it is now).

In your example the return statement is essentially an unreachable code whereas after the fix is applied the method will return 1, which means that the semantics has changed.

**Comment by jamesHargreaves12** [comment]
@JosefPihrt The `return 1` statement is reachable in the first example. https://dotnetfiddle.net/o1m9ZH

Adding a label doesn't change 'normal' sequential program execution. It just enables you to use the goto syntax.

**Comment by jamesHargreaves12** [comment]
Ok glad we are one the same page :) - I believe this contains the correct fix for this issue https://github.com/JosefPihrt/Roslynator/pull/1031

**Comment by JosefPihrt** [comment]
fixed with #1031
